In [12]:
import requests
import os
import json

In [3]:
response = requests.get("http://neuromorpho.org/api/neuron/id/2245.json")

In [17]:
response.status_code

200

In [16]:
"http://neuromorpho.org/api/neuron/id/%d.json" % 5

'http://neuromorpho.org/api/neuron/id/5.json'

# Attempt to download all neuron metadata

In [30]:
#neurons_allrequest = dict()
#neurons_request = dict()
# total 70465 id's, but some of them are deleted. 
# total 70024 reconstructed neurons
for i in range(1,70465):
    url = "http://neuromorpho.org/api/neuron/id/%d.json" % i
    neuron = requests.get(url)
    if neuron.status_code != 200:
        print('%i: bad request status' % i)
    if i % 100 == 0:
        print(i)
    neurons_allrequest[i] = neuron.json()
    neurons_request[i] = neuron.status_code

42700
42800
42900
43000
43100
43200
43300
43400
43500
43600
43700
43800
43900
44000
44100
44200
44300
44400
44500
44600
44700
44800
44900
45000
45100
45200
45300
45400
45500
45600
45700
45800
45900
46000
46100
46200
46300
46400
46500
46600
46700
46800
46900
47000
47100
47200
47300
47400
47500
47600
47700
47800
47900
48000
48100
48200
48300
48400
48500
48600
48700
48800
48900
49000
49100
49200
49300
49400
49500
49600
49700
49800
49900
50000
50100
50200
50300
50400
50500
50600
50700
50800
50900
51000
51100
51200
51300
51400
51500
51600
51700
51800
51900
52000
52100
52200
52300
52400
52500
52600
52700
52800
52900
53000
53100
53200
53300
53400
53500
53600
53700
53800
53900
54000
54100
54200
54300
54400
54500
54600
54700
54800
54900
55000
55100
55200
55300
55400
55500
55600
55700
55800
55900
56000
56100
56200
56300
56400
56500
56600
56639: bad request status
56700
56800
56900
57000
57100
57153: bad request status
57200
57300
57400
57500
57600
57700
57800
57900
58000
58100
58200
58300
58400


In [33]:

# save file
with open('../data/neurons_allrequest.json', 'w') as f:
    json.dump(neurons_allrequest, f)

In [32]:
len(neurons_allrequest)

70464

# exclude neurons with failed api requests (no reconstruction)

In [39]:
neurons_all = {i: neurons_allrequest[i] for i in neurons_allrequest if not 'error' in neurons_allrequest[i]}

In [40]:
len(neurons_good)

70024

In [41]:
with open('../data/neurons_all.json', 'w') as f:
    json.dump(neurons_all, f)

# example neuron

In [42]:
neuron.json()

{'_links': {'measurements': {'href': 'http://neuromorpho.org/api/morphometry/id/70464'},
  'self': {'href': 'http://neuromorpho.org/api/neuron/id/70464'}},
 'age_classification': 'adult',
 'archive': 'Johnson_T',
 'attributes': 'No Diameter, 2D, Angles',
 'brain_region': ['retina', 'inner plexiform layer', 'right', 'nasal'],
 'cell_type': ['principal cell', 'ganglion'],
 'corrected_value': None,
 'corrected_xy': None,
 'corrected_z': None,
 'deposition_date': '2016-01-04',
 'domain': 'Dendrites, No Soma, No Axon',
 'experiment_condition': 'Control',
 'magnification': '20',
 'max_age': '3.0',
 'max_weight': None,
 'min_age': '2.0',
 'min_weight': None,
 'neuron_id': 70464,
 'neuron_name': 'Ms504_RT_NAS_C4_Gr2_D7',
 'note': 'Organotypic tissue culture imaged at day 7. A reconstruction of this neuron immediatey after dissection is also available <a href=" dableFiles/johnson_t/Additional/culture0.zip">here</a>.',
 'objective_type': 'dry',
 'original_format': 'NeuronJ.ndf',
 'physical_Integ

# Download images

In [43]:
import urllib

In [ ]:
download_success = dict()
for i in neurons_all:
    if i % 100 == 0:
        print(i)
    url = neurons_good[i]['png_url']
    download_success[i] = True
    try:
        urllib.request.urlretrieve(url, "../data/png/%d.png" % i)
    except:
        download_success[i] = False

100
200
300
400
500
600
700
800
900
1000
1100
1200
1300
1400
1500
1600
1700
1800
1900
2000
2100
2200
2300
2400
2500
2600
2700
2800
2900
3000
3100
3200
3300
3400
3500
3600
3700
3800
3900
4000
4100
4200
4300
4400
4500
4600
4700
4800
4900
5000
5100
5200
5300
5400
5500
5600
5700
5800
5900
6000
6100
6200
6300
6400
6500
6600
6700
6800
6900
7000
7100
7200
7300
7400
7500
7600
7700
7800
8000
8100
8200
8300
8400
8500
8600
8700
8800
8900
9000
9100
9200
9300
9400
9500
9600
9700
9800
9900
10000
10100
10200
10300
10400
10500
10600
10700
10800
10900
11000
11100
11200
11300
11400
11500
11600
11700
11800
11900
12000
12100
12200
12300
12400
12500
12600
12700
12800
12900
13000
13100
13200
13300
13400
13500
13600
13700
13800
13900
14000
14100
14200
14300
14400
14500
14600
14700
14800
14900
15000
15100
15200
15300
15400
15500
15600
15700
15800
15900
16000
16100
16200
16300
16400
16500
16600
16700
16800
16900
17000
17100
17200
17300
17400
17500
17600
17700
17800
17900
18000
18100
18200
18300
18400
18500
186

In [13]:
with open('../data/neurons_all.json', 'r') as f:
    neurons_all = json.load(f)

In [18]:
neuron_ids = sorted([int(i) for i in neurons_all])
len(neuron_ids)

70024

In [14]:
png_links = [neurons_all[i]['png_url'] for i in neurons_all]

In [15]:
png_links

['http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_001.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_002.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_003.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_004.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_005.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_006.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_038.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_039.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_040.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_041.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_042.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_043.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_044.png',
 'http://neuromorpho.org/images/imageFiles/Wearne_Hof/cnic_045.png',
 'http://neuromorpho.org/images/im

In [7]:
png_all = sorted([int(x.strip('.png')) for x in os.listdir('../data/png/')])

In [9]:
len(png_all)

68715

In [29]:
neurons_no_png = list(set(neuron_ids) - set(png_all))
len(neurons_no_png)

1309

In [32]:
neurons_no_png

[70255,
 62687,
 62688,
 62689,
 62522,
 62690,
 62733,
 62929,
 62691,
 60656,
 62692,
 60657,
 62725,
 62693,
 60658,
 62694,
 60659,
 62896,
 62695,
 60660,
 62930,
 62696,
 60661,
 62697,
 60662,
 62698,
 60663,
 62699,
 60664,
 62700,
 60665,
 62931,
 62701,
 60666,
 62702,
 60667,
 62703,
 60668,
 62704,
 60669,
 62705,
 60670,
 62932,
 62706,
 60671,
 70259,
 62707,
 60672,
 62708,
 62709,
 62710,
 62933,
 62711,
 62712,
 62713,
 62714,
 62527,
 62715,
 62934,
 62716,
 62717,
 34972,
 34973,
 34974,
 34975,
 34976,
 34977,
 34978,
 34979,
 34980,
 34981,
 34982,
 34983,
 34984,
 34985,
 34986,
 34987,
 34988,
 34989,
 34990,
 34991,
 34992,
 34993,
 34994,
 34995,
 34996,
 34997,
 34998,
 34999,
 35000,
 35001,
 35002,
 35003,
 35004,
 35005,
 35006,
 35007,
 35008,
 35009,
 35010,
 35011,
 35012,
 35013,
 35014,
 35015,
 35016,
 35017,
 35018,
 35019,
 35020,
 35021,
 35022,
 35023,
 35024,
 35025,
 35026,
 35027,
 35028,
 35029,
 35030,
 35031,
 35032,
 35033,
 35034,
 35035,


In [31]:
bad_png_links = [neurons_all[str(i)]['png_url'] for i in neurons_no_png]
bad_png_links

['http://neuromorpho.org/images/imageFiles/Allen Cell Types/Scnn1a-Tg3-Cre_Ai14-241666-06-02-01_527110314_m.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_4-21_c1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_4-9_c1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_5-10_c1.png',
 None,
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_5-15_c1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/VM_133_c1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/VM17_2_cell1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_5-22_c1.png',
 'http://neuromorpho.org/images/imageFiles/Radley-4/SM102-1-B.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_5-2_c1.png',
 'http://neuromorpho.org/images/imageFiles/Radley-4/SM102-12-A.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_9-23_c1.png',
 'http://neuromorpho.org/images/imageFiles/Briggs-Bis/Bub_5-7_c3.png',
 'http://neuromorpho.org/i

# Download images with bad links in json metadata

## by scrapping the image file from html webpage

In [62]:
import urllib3
from bs4 import BeautifulSoup
import re
import urllib

In [34]:
http = urllib3.PoolManager()

In [65]:
download_success = dict()
prefix = 'http://neuromorpho.org/'
for cid in neurons_no_png:
    cid = str(cid)
    #if int(cid) % 100 == 0:
    #    print(cid)
    http_address = 'http://neuromorpho.org/neuron_info.jsp?neuron_id=%s' % cid
    html = http.request('GET', http_address)
    soup = BeautifulSoup(html.data)
    imgfile = soup.find_all('img', src=re.compile('^./images/imageFiles'))
    if len(imgfile) != 1:
        print('image link error: %d image links found' % len(imgfile))
    sublink = imgfile[0].get('src')
    sublink = sublink.lstrip('.')
    sublink = sublink.lstrip('/')
    url = prefix + sublink
    url = url.replace(' ', '%20')
    print(url)
    download_success[cid] = True
    try:
        urllib.request.urlretrieve(url, "../data/png/%s.png" % cid)
    except:
        print('%s: download error.' % cid)
        download_success[cid] = False

/home/yueqi/anaconda/envs/py36/lib/python3.6/site-packages/bs4/__init__.py:181: UserWarning: No parser was explicitly specified, so I'm using the best available HTML parser for this system ("lxml"). This usually isn't a problem, but if you run this code on another system, or in a different virtual environment, it may use a different parser and behave differently.

The code that caused this warning is on line 193 of the file /home/yueqi/anaconda/envs/py36/lib/python3.6/runpy.py. To get rid of this warning, change code that looks like this:

 BeautifulSoup(YOUR_MARKUP})

to this:

 BeautifulSoup(YOUR_MARKUP, "lxml")

  markup_type=markup_type))


http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre_Ai14-241666-06-02-01_527110314_m.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_4-21_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_4-9_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_5-10_c1.png
http://neuromorpho.org/images/imageFiles/Leroy/28-08-MN4.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_5-15_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_133_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM17_2_cell1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_5-22_c1.png
http://neuromorpho.org/images/imageFiles/Radley/SM102-1-B.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_5-2_c1.png
http://neuromorpho.org/images/imageFiles/Radley/SM102-12-A.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_9-23_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_5-7_c3.png
http://neuromorpho.org/images/imageFiles/Radley/SM102-2-A.png
ht

http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-19.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-21.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-24.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-27.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-29.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-33.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-39.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-40.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-41.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-48.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-53.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-59.png
http://neuromorpho.org/images/imageFiles/De%20Paola/Dataset1-Edited-70.png
http://neuromorpho.org/im

http://neuromorpho.org/images/imageFiles/Briggs/VM_158_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_189_c2_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_193_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_205_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM22_2_Cell2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_247_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_257_c2_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-167-3.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_274_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-173-1.png
62900
http://neuromorpho.org/images/imageFiles/Briggs/VM18_4_Cell1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_281_c1_V2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_282_c1_V2.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Ctgf-2A-dgCre_Ai14_IVSCC_-229234-03-01-01_541532414_m.png
http://neuromor

http://neuromorpho.org/images/imageFiles/Radley/SM35-1_6.png
http://neuromorpho.org/images/imageFiles/Briggs/VM24_3_Cell2.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-212873-04-01-01_510987644_m.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-159-5.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-230822-04-02-01_519262717_m.png
http://neuromorpho.org/images/imageFiles/Radley/SM35-2_1.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg2-Cre_Ai14-245347-05-01-01_537047525_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre_Ai14-187849-06-01-01_512328420_m.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-160-1.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-171-3.png
http://neuromorpho.org/images/imageFiles/Radley/SM35-2_2.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Sst-IRES-Cre_A

http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-163-4.png
http://neuromorpho.org/images/imageFiles/Radley/RG123-3.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-163-5.png
http://neuromorpho.org/images/imageFiles/Radley/RG123-5.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-163-6.png
http://neuromorpho.org/images/imageFiles/Radley/RG123-6.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-176-2.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-163-7.png
http://neuromorpho.org/images/imageFiles/Radley/RG123-7.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-192794-05-02-01_491770902_m.png
http://neuromorpho.org/images/imageFiles/Radley/RG123-9.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-164-2.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_9-22_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-180-1.png
http://neuromorp

http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-185555-03-01-01_491119960_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-185555-05-01-01_491119381_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-185555-05-02-01_492413487_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-185556-06-01-01_492413499_m.png
http://neuromorpho.org/images/imageFiles/Radley/RG135-10.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-192819-04-02-01_491119743_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-192821-05-01-01_491771409_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-195039-04-02-01_491119581_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Htr3a-Cre_NO152_Ai14-198254-03-02-01_491119526_m.png
http://neuromorpho.

http://neuromorpho.org/images/imageFiles/Radley/SM41-4_1.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197332-04-02-01_492100936_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197332-06-02-01_491771446_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197352-06-01-01_491253869_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197353-06-02-01_496079587_m.png
http://neuromorpho.org/images/imageFiles/Radley/sm41-4_2.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197851-04-01-01_491770985_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-197851-05-02-01_492413372_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb-IRES2-Cre-D_Ai14-212872-05-01-01_496164023_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rorb

http://neuromorpho.org/images/imageFiles/Lambe/C16---Adult-WT-Nic.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Sst-IRES-Cre-Ai14-348583888.png
http://neuromorpho.org/images/imageFiles/Radley/SM42-2_7.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_9_9_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM17_3_cell1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM17_4_Cell1.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Sst-IRES-Cre-Ai14-471774606.png
http://neuromorpho.org/images/imageFiles/Radley/SM42-2_8.png
http://neuromorpho.org/images/imageFiles/Briggs/bubbles-R-LGN-173-2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM18_2_Cell1.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Rbp4-Cre_KL100_Ai14-205726-04-01-01_514342418_m.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Sst-IRES-Cre-Ai14-469822422.png
http://neuromorpho.org/images/imageFiles/Radley/SM43-1_1.png
http://neuromorpho.or

http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre-Ai14-328933585.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre-Ai14-470986539.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre-Ai14-473845048.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre-Ai14-475244991.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg3-Cre-Ai14-475124505.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Sst-IRES-Cre-Ai14-475332668.png
http://neuromorpho.org/images/imageFiles/Guerra%20da%20Rocha/cc05lam03cel01pa.png
http://neuromorpho.org/images/imageFiles/Guerra%20da%20Rocha/cc06lam06cel04pa.png
http://neuromorpho.org/images/imageFiles/Guerra%20da%20Rocha/cc07lam03cel01.png
http://neuromorpho.org/images/imageFiles/Guerra%20da%20Rocha/cc07lam03cel02.png
http://neuromorpho.org/images/imageFiles/Guerra%20da%20Rocha/cc07lam06cel01pa.png
http://neuromorpho.o

http://neuromorpho.org/images/imageFiles/Radley/SM36-2_4.png
http://neuromorpho.org/images/imageFiles/Radley/SM36-2_5.png
http://neuromorpho.org/images/imageFiles/Radley/SM36-2_6.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-1_2.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-1_3.png
http://neuromorpho.org/images/imageFiles/Radley/Sm63-1_4.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-1_5.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-1_6.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_1.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_2.png
60700
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_3.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_5.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_6.png
http://neuromorpho.org/images/imageFiles/Radley/SM63-3_7.png
http://neuromorpho.org/images/imageFiles/Radley/SM86-1.png
http://neuromorpho.org/images/imageFiles/Radley/SM86-10.png
http://neuromorpho.or

http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM46-3.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM46-2.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM46-1.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM45-2.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM45-1.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM44-4-Slice-2.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM44-3-Slice-1.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM44-2-Slice-1.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM44-1-Slice-1.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM43-2-Slice-3.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM43-1-Slice-3.png
http://neuromorpho.org/images/imageFiles/De%20Koninck/Fr-LY-Pyr-OM39-2.png
http://neuromorpho.org/images/imageFiles/De%20Koninc

http://neuromorpho.org/images/imageFiles/Radley/SM71-3.png
http://neuromorpho.org/images/imageFiles/Radley/SM71-4.png
http://neuromorpho.org/images/imageFiles/Radley/SM71-5.png
http://neuromorpho.org/images/imageFiles/Radley/SM71-6.png
http://neuromorpho.org/images/imageFiles/Radley/SM71-8.png
http://neuromorpho.org/images/imageFiles/Radley/SM71-9.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-1-A.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-1-C.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-2-B.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-2-C.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-3-B.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-3-C.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-4-B.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-4-C.png
http://neuromorpho.org/images/imageFiles/Radley/SM73-5-B.png
http://neuromorpho.org/images/imageFiles/Radley/SM75-1-A.png
http://neuromorpho.org/images/imageF

http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-1_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-21_c1.png
http://neuromorpho.org/images/imageFiles/Bullmann/44_1_4.png
51085: download error.
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-2_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-6_c1.png
http://neuromorpho.org/images/imageFiles/Briggs/VM_129_c2.png
http://neuromorpho.org/images/imageFiles/Briggs/VM26_4_Cell3.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-7_c2.png
http://neuromorpho.org/images/imageFiles/Allen%20Cell%20Types/Scnn1a-Tg2-Cre_Ai14-215161-05-01-01_510985337_m.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-7_c3.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10-8_c2.png
http://neuromorpho.org/images/imageFiles/Briggs/Bub_10_11_c1.png
http://neuromorpho.org/images/imageFiles/Capogna/Bienvenu_etal_JN15_1.png
http://neuromorpho.org/images/imageFiles/Capogna/Bienvenu_etal_JN15_2.png
http://neurom

## one download error -- 51085. the webpage does not show image file either -- bad link. 

# morphology data

In [7]:
morph = requests.get("http://neuromorpho.org/api/morphometry/id/2245.json")

In [8]:
morph.json()

{'bif_ampl_local': 73.53,
 'branch_Order': 19.0,
 'contraction': 0.77,
 'depth': 36.99,
 'diameter': 0.24,
 'eucDistance': 550.73,
 'fractal_Dim': 1.05,
 'fragmentation': 7096.0,
 'height': 583.85,
 'length': 12566.7,
 'n_bifs': 142.0,
 'n_branch': 285.0,
 'n_stems': 1.0,
 'neuron_id': 2245,
 'neuron_name': 'AM48-2',
 'partition_asymmetry': 0.56,
 'pathDistance': 1015.12,
 'pk_classic': 2.0,
 'soma_Surface': None,
 'surface': 9475.06,
 'volume': 568.5,
 'width': 396.67}

In [ ]:
morpho_allrequest = dict()
morpho_request = dict()
for i in range(1,70465):
    url = "http://neuromorpho.org/api/morphometry/id/%d.json" % i
    morpho = requests.get(url)
    if morpho.status_code != 200:
        print('%i: bad request status' % i)
    if i % 100 == 0:
        print(i)
    try:
        morpho_allrequest[i] = morpho.json()
    except:
        morpho_allrequest[i] = None
    morpho_request[i] = morpho.status_code

In [ ]:
with open('../data/morpho_allrequest.json', 'w') as f:
    json.dump(morpho_allrequest, f)

In [ ]:
morpho_all = {i: morpho_allrequest[i] for i in morpho_allrequest if not morpho_allrequest[i] is None}

In [ ]:
len(morpho_all)

In [ ]:
with open('../data/morpho_all.json', 'w') as f:
    json.dump(morpho_all, f)

# literature info

In [76]:
pmid_all = []
for i in [0,1,2,3]:
    pmid_request = requests.get("http://neuromorpho.org/api/literature/fields/PMID.json?page=%d" % i)
    pmid_chunk = pmid_request.json()
    pmid_chunk = pmid_chunk['fields']
    pmid_all.extend(pmid_chunk)

In [78]:
len(pmid_all)

1874

In [79]:
 with open('../data/literature.json', 'w') as f:
    json.dump(pmid_all, f)